src/Lib/Printer.hs

```diff
@@ -214,6 +216,47 @@ instance Printer AnsiPrinter where
   setUnderline p u      = unit $ ansiSetConsole p (\con -> con{ underline = u })
 
 
+data AnsiStringPrinter = AnsiString (Var AnsiConsole) (Var String)
```

Comment:
Added a printer that prints color, but to a string instead of IO, so that we can show colored output in the language server.